evmcore/tx_validation_test.go

```diff
@@ -18,151 +18,131 @@ package evmcore
 
 import (
 	"fmt"
-	"math"
 	"math/big"
 	"testing"
 
+	"github.com/0xsoniclabs/sonic/inter/state"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core"
 	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/crypto/kzg4844"
 	"github.com/ethereum/go-ethereum/params"
 	"github.com/holiman/uint256"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
 )
 
-// getTestPoolOptions returns a set of options to adjust the validation of transactions
-// considering them as local transactions with a min tip of 1.
-func getTestPoolOptions() poolOptions {
-	return poolOptions{
-		minTip:       big.NewInt(1),
-		isLocal:      true,
-		currentState: nil,
-		locals:       nil,
-	}
-}
-
-// getTestBlockState returns a test set of base feee and max gas that can be used for
-// validating transactions in the context of a block. It initializes the state
-// with a base fee of 1 and a max gas of 100_000
-func getTestBlockState() blockState {
-	return blockState{
-		baseFee: big.NewInt(1),
-		maxGas:  100_000,
-	}
-}
-
-// getTestNetworkRules returns a set of network rules to adjust the validation of transactions
-// so that it accepts all types of transactions. It also sets the signer to a new Prague signer with chain ID 1.
-func getTestNetworkRules() NetworkRules {
-	return NetworkRules{
-		istanbul: true,
-		eip2718:  true,
-		eip1559:  true,
-		shanghai: true,
-		eip4844:  true,
-		eip7623:  true,
-		eip7702:  true,
-		osaka:    true,
-		signer:   types.NewPragueSigner(big.NewInt(1)),
-		// EIP-7825 fixes this limit to 16M gas, Sonic network leaves this
-		// parameter configurable via network rules
-		maxTxGas: 16_000_000,
-	}
-}
-
 ////////////////////////////////////////////////////////////////////////////////
 // Static Validation
 
 func TestValidateTxStatic_Value_RejectsTxWithNegativeValue(t *testing.T) {
-	for name, tx := range getTxsOfAllTypes() {
-		t.Run(name, func(t *testing.T) {
-			if isBlobOrSetCode(tx) {
-				t.Skip("blob and setCode transactions cannot have negative value because they use uint256 Value")
-			}
-			setValueToNegative(t, tx)
+	txs := []types.TxData{
```

Comment:
would it make sense to add a comment in each tests as for why there are some types of transactions not tested for that property ??